src/app/shared/loan-card/loan-card.component.ts

```diff
@@ -36,6 +36,11 @@ export class LoanCardComponent implements OnInit {
       this.rightValue = this.formatAmount(this.loan.pendingAmount);
       this.durationLabel = 'Remaining';
       this.durationValue = Utils.formatDelta(this.loan.remainingTime);
+      const timeLeft = Utils.formatDelta(this.loan.remainingTime);
+      if (timeLeft.startsWith('-')) {
```

Comment:
Consider checking if the remaining time is negative or if `now > debt time`, avoid checking conditions from already formatted strings.